app/include/zmk/display.h

```diff
@@ -6,4 +6,48 @@
 
 #pragma once
 
-int zmk_display_init();
\ No newline at end of file
+struct k_work_q *zmk_display_work_q();
+
+bool zmk_display_is_initialized();
+int zmk_display_init();
+
+/**
+ * @brief Macro to define a ZMK event listener that handles the thread safety of fetching
+ * the necessary state from the system work queue context, invoking a work callback
+ * in the display queue context, and properly accessing that state safely when performing
+ * display/LVGL updates.
+ * @param listener THe ZMK Event manager listener name
+ * @param state_type The struct/enum type used to store/transfer state.
+ * @param cb The callback to invoke in the dispaly queue context to update the UI
```

Comment:
Would be helpful to list the expected function signatures for `cb` and `state_func`.

Reply:
Added, as best I could w/ Doxygen syntax.